### ISSUE TITLE
Rundeck performance

### ISSUE THREAD

**Issue by kino71** [rundeck/rundeck#1545]
Hello,
I'm migrating our rundeck installation from launcher 2.5.0 with everything on file system, to 2.5.2 on Oracle.
Configuration and migration of projects and jobs is fine.

It takes more than expected to open the job page of a project.
As soon as I click on the project name, there are a number of evaluation (I think one for every project of the main page), the last is :

2015-11-12 12:31:03,669 INFO  SAREAuthorization - Evaluating Decision for: res<name:PRJ_TEST, type:project> subject.......  GRANTED_ACTIONS_AND_COMMANDS_MATCHED (0ms)

then there is a pause of 5 secs and the log starts again:

2015-11-12 12:31:08,887 INFO  SAREAuthorization - Evaluating Decision for: res<name:Deploy startup scripts, type:job.......GRANTED_ACTIONS_AND_COMMANDS_MATCHED (0ms)
followed by many evaluations.

Most of the time is spent between those two lines.
How can I understand exactly where the bottleneck is ?

Thanks
Francoi

**Comment by kino71** [comment]
Hello,
I tried the same project/job configuration with the default H2DB :

2015-11-12 17:33:18,576 [qtp11394340-117] INFO  com.dtolabs.rundeck.core.authorization.providers.SAREAuthorization - Evaluating Decision for: res<name:PRJ_TEST,  .................  GRANTED_ACTIONS_AND_COMMANDS_MATCHED (0ms)

the pause here is about 3 secs 

2015-11-12 17:33:21,487 [qtp11394340-117] INFO  com.dtolabs.rundeck.core.authorization.providers.SAREAuthorization - Evaluating De
cision for: res<name:Deploy startup scripts,  .........  GRANTED_ACTIONS_AND_COMMANDS_MATCHED (0ms)

but overall, it seems much faster with H2DB than Oracle.

Any thoughts ?

Franco

**Comment by kino71** [comment]
Hello,
I couldn't find the reason why the GUI was so slow using Oracle as db, so I tried it on Sybase ASE 16. 

GUI performance is better, it might be interesting to someone else the configuration for Sybase :

You'll need to copy jconn4.jar in server/lib, then in rundeck-config.properties :

dataSource.url = jdbc:sybase:Tds:[db_host]:4321/db_mercury?HOSTNAME=[hostname]&DB_NAME=/[db_name]
dataSource.driverClassName=com.sybase.jdbc4.jdbc.SybDriver
dataSource.username = [user for db ]
dataSource.password = [password for db]
dataSource.dialect = org.hibernate.dialect.SybaseASE157Dialect

I believe that HOSTNAME and DB_NAME in dataSource.url aren't actually used.

The creation of one table (storage) will fail, I've manually created it with:

create table storage (id bigint identity not null, version bigint not null, data image null, date_created datetime not null, dir varchar(2048) null, json_data text null, last_updated datetime not null, name varchar(1024) not null, namespace varchar(255) null, path_sha varchar(40) not null, primary key (id)) lock datarows

It seems to work for the time being.

I'm also running with  -Dhibernate.jdbc.use_get_generated_keys=true. It has been inherited by the previous attempt with Oracle, I'm not sure it is needed.

Obviously  we also have rundeck.projectsStorageType=db

As I mentioned, GUI performance is better compared with Oracle. This might be a problem with the connection between rundeck and oracle host, since we can't see any slow query, there are just lots of them .

Franco
